ir/instr.cpp

```diff
@@ -169,12 +169,44 @@ static expr fm_poison(const expr &a, const expr &b, const expr &val,
   return ret;
 }
 
+static StateValue vectorBinOp(const StateValue &av,
+                              const StateValue &bv, BinOp::Op op,
+                              const AggregateType *aty) {
+  expr res_val;
+  expr res_np;
+  for (unsigned i = 0 ; i < aty->getNumElements(); ++i) {
+    auto a = aty->extract(av, i);
+    auto b = aty->extract(bv, i);
+    expr val;
+    expr non_poison = a.non_poison && b.non_poison;
+
+    switch (op) {
```

Comment:
Note there would be a lot of code duplication, between vector and non-vector paths. I will wrap the code with the same shapes with macros.